### DISCUSSION TITLE
One external database, using multiple Nominati(?) versions

### DISCUSSION THREAD

**Discussion by akobelan** [osm-search/Nominatim#3264] [General]
Hi,

I loaded an external  Postgres database with OSM using Nominatim 4.2. I installed multiple Nominatim instances querying that single instance of Postgres. I proceeded to replace the 4.2 instances with 4.3 instances. Not all of them at once so I have an equal number of 4.2 and 4.3 instances.

Should I expect Nominatim 4.3 to seamlessly work alongside the 4.2 clients?

**Comment by mtmail** [comment]
Part of the [migration](https://nominatim.org/release-docs/latest/admin/Migration/) is updating PG/PSQL functions in the database. (The files you see in `lib-sql/functions`). So the Python code of the app and the PSQL functions in the database are developed together and need to match. Running two versions of Nominatim against the same database instance won't work (if it does it's a coincidence).

**Comment by akobelan** [comment]
Got it. Running the same version would work, then? I guess I need to verify if the data in Postgres is aligned with my Nominatim instance. Is there a table that keeps track of this information? The information being what versions I should be running.

**Reply by mtmail** [reply]
`nominatim admin --check-database` will soon have such a check https://github.com/osm-search/Nominatim/pull/3238

**Reply by mtmail** [reply]
```
select * from nominatim_properties where property='database_version';
     property     |  value
------------------+---------
 database_version | 4.3.2-0
```
